pandas/_libs/tslibs/src/datetime/np_datetime.c

```diff
@@ -378,11 +379,14 @@ int convert_pydatetime_to_datetimestruct(PyObject *dtobj,
             if (tmp == NULL) {
                 return -1;
             }
-            seconds_offset = PyInt_AsLong(tmp);
+            tmp_int = PyNumber_Long(tmp);
```

Comment:
You will also want to do a `if (tmp_int == NULL)` check and return an error / DECREF tmp if that happens. Otherwise lgtm